src/main/scala/pipedsl/typechecker/TypeInferenceWrapper.scala

```diff
@@ -756,7 +840,9 @@ object TypeInferenceWrapper
       TFun(List(TSizedInt(TBitWidthLen(t.addrSize), sign = TUnsigned())), t.elem)
      }
 
-    private def substIntoCall(expectedType: TFun, ca: ECall, args: List[Expr], env: TypeEnv) = {
+    private def substIntoCall(expectedT: TFun, ca: ECall, args: List[Expr], env: TypeEnv) = {
+     //val expectedType = degenerify(expectedT).asInstanceOf[TFun]
```

Comment:
delete? why do we not need this line?

Reply:
turns out it was actually helpful to know which of these used to be generics. Old code gone now